File: pageindicatorview/src/main/java/com/rd/draw/drawer/type/DropDrawer.java
package com.rd.draw.drawer.type;

import android.graphics.Canvas;
import android.graphics.Paint;
import androidx.annotation.NonNull;
import com.rd.animation.data.Value;
import com.rd.animation.data.type.DropAnimationValue;
import com.rd.draw.data.Indicator;
import com.rd.draw.data.Orientation;

public class DropDrawer extends BaseDrawer {

    public DropDrawer(@NonNull Paint paint, @NonNull Indicator indicator) {
        super(paint, indicator);
    }

    public void draw(
            @NonNull Canvas canvas,
            @NonNull Value value,
            int coordinateX,
            int coordinateY) {

        if (!(value instanceof DropAnimationValue)) {
            return;
        }

        DropAnimationValue v = (DropAnimationValue) value;
        int unselectedColor = indicator.getUnselectedColor();
        int selectedColor = indicator.getSelectedColor();
        float radius = indicator.getRadius();

        paint.setColor(unselectedColor);
        canvas.drawCircle(coordinateX, coordinateY, radius, paint);

        paint.setColor(selectedColor);
        if (indicator.getOrientation() == Orientation.HORIZONTAL) {
            canvas.drawCircle(v.getWidth(), v.getHeight(), v.getRadius(), paint);
        } else {
            canvas.drawCircle(v.getHeight(), v.getWidth(), v.getRadius(), paint);
        }
    }
}

File: pageindicatorview/src/main/java/com/rd/draw/drawer/type/DropDrawer.kt
package com.rd.draw.drawer.type

import android.graphics.Canvas
import android.graphics.Paint
import com.rd.animation.data.Value
import com.rd.animation.data.type.DropAnimationValue
import com.rd.draw.data.Indicator
import com.rd.draw.data.Orientation

class DropDrawer(paint: Paint, indicator: Indicator) : BaseDrawer(paint, indicator) {

	fun draw(
			canvas: Canvas,
			value: Value,
			coordinateX: Int,
			coordinateY: Int) {

		if (value !is DropAnimationValue) {
			return
		}

		val unselectedColor = indicator.unselectedColor
		val selectedColor = indicator.selectedColor
		val radius = indicator.radius.toFloat()

		paint.color = unselectedColor
		canvas.drawCircle(coordinateX.toFloat(), coordinateY.toFloat(), radius, paint)

		paint.color = selectedColor
		if (indicator.orientation === Orientation.HORIZONTAL) {
			canvas.drawCircle(value.width.toFloat(), value.height.toFloat(), value.radius.toFloat(), paint)
		} else {
			canvas.drawCircle(value.height.toFloat(), value.width.toFloat(), value.radius.toFloat(), paint)
		}
	}
}
